rasa/core/domain.py

```diff
@@ -639,103 +624,109 @@ def input_states(self) -> List[Text]:
         """Returns all available states."""
 
         return (
-            self.intent_states
-            + self.entity_states
+            self.intents
+            + self.entities
             + self.slot_states
-            + self.prev_action_states
-            + self.form_states
+            + self.action_names
+            + self.form_names
         )
 
-    def get_parsing_states(self, tracker: "DialogueStateTracker") -> Dict[Text, float]:
+    def _get_user_states(
+        self, tracker: "DialogueStateTracker"
+    ) -> Dict[Text, Dict[Text, Union[Text, Tuple[Text]]]]:
         state_dict = {}
 
         # Set all found entities with the state value 1.0, unless they should
         # be ignored for the current intent
         latest_message = tracker.latest_message
 
-        if not latest_message:
+        if not latest_message or latest_message == UserUttered.empty():
```

Comment:
this would avoid creating an additional object just to check if it's empty
```suggestion
        if not latest_message or latest_message.is_empty():
```